skimage/restoration/tests/test_restoration.py

```diff
@@ -87,6 +87,18 @@ def test_richardson_lucy():
     np.testing.assert_allclose(deconvolved, np.load(path), rtol=1e-3)
 
 
+def test_gaussian_natural_prior():
+    test_img_astro = rgb2gray(astronaut())
+    psf = np.ones((5, 5)) / 25
+    data = convolve2d(test_img_astro, psf, 'same')
+    np.random.seed(0)
+    data += 0.1 * data.std() * np.random.standard_normal(data.shape)
+    deconvolved = restoration.gaussian_natural_prior(data, psf)
+
+    path = image_fetcher.fetch('restoration/tests/astronaut_dnp.npy')
```

Comment:
How was this file generated? Was it saved using the implementation in this PR or was it created with a 3rd party implementation such as the Matlab code of the original authors?

If it is generated in the code in this PR, then this doesn't really test that the output here is valid, but will help ensure that any future changes keep output that is consistent with the current implementation.

Reply:
It is probably worth adding another test case similar to the above but where the original image is compared based on some metric to determine if it has improved. `structural_similarity` may be a good candidate for this. e.g. something like

```Python
from skimage.metrics import structural_similarity
ideal = rgb2gray(astronaut())
# CODE AS ABOVE TO DO THE DECONVOLUTION
ssim_blurred = structural_similarity(ideal, data)
ssim_deconv = structural_similarity(ideal, deconvoled)
assert ssim_deconv > ssim_blurred
```

Reply:
@grlee77 The file was created with a one to one translation from Matlab to Python based on the reference implementation from the original authors, referenced as [1] in the code description. I have checked the translated code on an individual line basis to ensure the results are the same in Matlab as well as in Python. So the image reflects the original code prior to our changes in the PR here.

Reply:
regarding the `structural_similarity` test, thats a great idea. However currently that test fails, probably due to either edge effects and/or the psf not being strong enough. Unfortunately I have no more time today to work on this, probably can take this on next week.

Reply:
@grlee77 The `structural_similarity` test works on data that has no noise added. I have implemented that test in the latest commit. Let me know if you are satisfied with this.